enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,458 @@
+---
+title: dummy-gateway-interface-ovn
+authors:
+- "@pliurh"
+reviewers:
+- "@jcaamano, OpenShift Networking"
+- "@pacevedom, Microshift"
+- "@pmtk, MicroShift"
+- "@trozet, OpenShift Networking"
+- "@zshi-redhat, MicroShift"
+approvers:
+- "@dhellmann"
+api-approvers:
+- None
+creation-date: 2023-04-26
+last-updated: 2023-04-27
+tracking-link:
+- https://issues.redhat.com/browse/NP-654
+see-also:
+- None
+replaces:
+- None
+superseded-by:
+- None
+---
+
+# Single Node OVN-Kubernetes for Microshift
+
+## Summary
+
+This enhancement proposes a new external gateway configuration of OVN-Kubernetes
+for Microshift. With the new configuration, a dummy interface will be created
+and the `br-ex` bridge will use it as the gateway interface.
+
+## Motivation
+
+Microshift is designed to run on IoT and edge devices, it can be deployed on
+various kinds of devices with different network environment. 
+
+Today configure-ovs.sh (microshift-ovs-init.service) sets up OVN-Kubernetes
+gateway bridge `br-ex` for OVN-Kubernetes.  There has been several types of
+issues associated with it mainly due to:
+
+1. transient network interruption when it moves the node IP from uplink to
+   gateway bridge
+
+2. certain type of network devices are not supported (e.g. wifi, tap/tun)
+
+3. it relies on (default) route info
+
+### User Stories
+
+* As an Microshift cluster administrator, I want the installation experience of
+Microshift is similar as a normal application. I don't want to have any
+networking interrupt during the installation.
+
+* As an Microshift cluster administrator, I want to use a specific network
+  devices (such as wifi, tap/tun) as the node default interface.
+
+* As an Microshift cluster administrator, I want to install Microshift on an
+  isolated host that has no default gateway available.
+
+### Goals
+
+* Decouple the br-ex bridge from the host physical interfaces.
+
+### Non-Goals
+
+* Creating a special gateway mode in OVN-Kubernetes for single node cluster.
+
+## Proposal
+
+Instead of creating the br-ex bridge with the node's default interface, we
+create a dummy interface and use it as the uplink of the br-ex. Since there is
+no real next-hop available via the dummy0 interface, we need to use the existing
+dummy next-hop masquerade address '169.254.169.4' as the default gateway. That
```

Comment:
This is done in every case, right? Not only when there is no default gateway. If that's the case, it might be worth clarifying it, along with the fact that it will not affect routing even if there is a default gateway due to be using local gateway mode.

Reply:
We will always use '169.254.169.4' as the default gateway for local gw.